projects/auth0-angular/src/lib/auth.client.ts

```diff
@@ -16,7 +16,7 @@ export class Auth0ClientFactory {
     const { redirectUri, clientId, maxAge, httpInterceptor, ...rest } = config;
 
     return new Auth0Client({
-      redirect_uri: redirectUri || window.location.origin,
+      redirect_uri: redirectUri || (typeof window !== 'undefined' ? window.location.origin : ''),
```

Comment:
Could you elaborate on what this change solves?

I dont see why you would want to create an `Auth0Client` instance without a `redirect_uri`. I think in cases where window is not available (like when using Server Side Rendering), it is advised to ensure a value is provided for `redirect_uri`. Checking if window is undefined and falling back to an empty string might fix errors on the server side, but will introduce other side effects to which you will need to workaround while it shouldn't have been needed in the first place.

Reply:
@frederikprijck hey sorry for delayed response - Usually any wide open global window usages just restrict the use cases in which the sdk can be used. For example when executed in a non-browser context this is the common blocker due to that sort of usage. The main problem this presents is that it causes an unnecessary wholesale blocker on the usage of this sdk in multiple contexts (even if used with a framework that allows those runtime overrides to be handled cleanly). As you mentioned there's other places in the auth0 stack that use global window however would love to help auth0 reduce those surface area restrictions.

There's lot of use cases to use this sdk provided a programmers ability to inject/provide for the things needed but unprotected global window access just prevent the broad use case applicability.
<img width="428" alt="Screen Shot 2021-04-13 at 6 05 10 PM" src="https://user-images.githubusercontent.com/457187/114639682-4c51a500-9c83-11eb-94b0-6366a9c13bd6.png">

Reply:
The vanilla sdk (the non-angular one provides this ability) however there are benefits in this sdk for usage with angular and just reducing window usage would increase it's applicability to more useful cases of use.

Reply:
Hey @NathanWalker, yes I am aware of the fact that using `window` is probably better avoided and won't benefit SSR in any way. However, what I can't seem to understand is what this change is actually solving other than not showing an error?

Imagine a site called: `'http://app.my-site.com'`, when initializing the SDK client side, we will use a `redirect_uri` that is set to 'http://app.my-site.com' (because of `window.location.origin`), while when initialized from the server side, and no window is available, the `redirect_uri` is now set to `''`, which won't put the SDK in a workable state, as the redirect_uri is something we probably want to have filled in.

The only way this would work is when specifying an actual `redirect_uri` when initializing the SDK, which would also take away the need to evaluate the right part of the `||`, so that wouldn't even need the check if I recall correctly?

Reply:
Perhaps a zoom call better to discuss sometime? Probably is a general look at auth0 stack and discussing use cases as to the why/how.

Reply:
Sure, happy to do so. I tried reaching out through Twitter and Discord, but that wasn't successful so far. Feel free to send me a DM on twitter or so to schedule something.